torch/testing/_legacy.py

```diff
@@ -1,13 +1,13 @@
-"""This module exist to be able to deprecate the dtype getters publicly without doing so internally. The deprecated
+"""This module exist to be able to deprecate functions publicly without doing so internally. The deprecated
```

Comment:
I'm a little confused about renaming this file and the structure here.

Why put get_all_device_types in this file?

Reply:
This file solves the issue of emitting a deprecation warning if the function is used from the public `torch.testing` namespace while still being able to use it without warning in internal use cases. Ideally, we would directly move the functionality to `torch.testing._internal` and import it to `torch.testing`, but this is not possible: since `torch.testing._internal` has some extra dependencies we cannot depend on it in `torch.testing`.